python/ray/tests/test_autoscaler.py

```diff
@@ -2239,14 +2239,13 @@ def testConfiguresNewNodes(self):
         )
 
         autoscaler.update()
+        # TODO(rueian): This is a hack to avoid running into race conditions
+        # within v1 autoscaler. These should no longer be relevant in v2.
+        self.waitForNodes(2)
```

Comment:
If this is needed for v2, can we put it under a `if` statement and only do it if we are testing v1?

Reply:
Hi @jjyao, these two tests utilize `MockAutoscaler`, which directly inherits from the v1 autoscaler, meaning that we only do the hack only for testing the v1 autoscaler already.